internal/worker/remoterelationconsumer/applicationworker.go

```diff
@@ -469,27 +449,70 @@ func (w *remoteApplicationWorker) handleRelationConsumption(
 
 	// We have not seen the relation before, or the relation was suspended, make
 	// sure it is registered (ack'd) on the offering side.
-	remoteRelation, err := w.registerRemoteRelation(
+	result, err := w.registerConsumerRelation(
 		ctx,
 		details.UUID, w.offerUUID, w.consumeVersion,
 		synthEndpoint, otherEndpointName,
 	)
 	if err != nil {
-		w.checkOfferPermissionDenied(ctx, err, "", "")
+		if statusErr := w.setApplicationOffererStatusMacaroonError(ctx, err); statusErr != nil {
+			w.logger.Errorf(ctx, "failed updating remote application %v status from remote model %v: %v", w.applicationName, w.offererModelUUID, statusErr)
+		}
 		return errors.Annotatef(err, "registering application %q and relation %q", w.applicationName, details.UUID)
 	}
-	w.logger.Debugf(ctx, "consumer relation registered for %q: %v", details.UUID, remoteRelation)
 
-	_ = remoteRelation
+	w.logger.Debugf(ctx, "consumer relation registered for %q: %v", details.UUID, result)
+
+	// Start the remote relation worker to watch for offerer relation changes.
+	// The aim is to ensure that we can track the suspended status of the
+	// relation so we can correctly react to that.
+	if err := w.createOffererRelationWorker(ctx, details.UUID, result.offererApplicationUUID, result.macaroon); err != nil {
+		return errors.Annotatef(err, "creating offerer relation worker for %q", details.UUID)
+	}
+
+	return nil
+}
+
+// Create a new worker to watch for changes to the relation in the offering
+// model.
+//
+// The worker is identified by the relation UUID, so it can track that
+// information, along with the offering application UUID and macaroon used to
+// access the relation in the offering model. If we have this information, we
+// should be able to pin point the relation in the offering model.
+func (w *remoteApplicationWorker) createOffererRelationWorker(
```

Comment:
`startOfferRelationWorker` is a bit more indicative I think. Or maybe `ensure...` given that existence is a no-op.